design/common/separate-git-repository-for-helm-charts.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Separate git Repository for Helm Charts
+target-version: release-1.6
+---
+
+# Separate git Repository for Helm Charts
+
+## Summary
+
+### Goals
+
+The goal of this design document / change is to use a separate git repository for Helm charts for the Rook ecosystem.
+The Rook ecosystem Helm charts include but not limited to the Rook operators and Rook custom resources.
+
+To know this goal has been achieved, a separate repository with the current Rook operator Helm charts and upcoming Helm charts should have been created.
+In addition to the repository, CI workflows should be established to be able to release the Helm charts when needed.
+
+### Non-Goals
+
+The design document is not about introducing new Helm Charts, but to move the Helm charts development to a separate repository.
+
+## Proposal details
+
+The proposal is to create a separate repository for the Helm charts.
+In the new repository for the Helm charts, a CI/CD workflow using GitHub Actions,
+utilizing the [GitHub chart-releaser Action](https://github.com/helm/chart-releaser-action) should be created to easily
+release changes to the Helm charts in the repository.
+
+Especially in regards to the idea of creating wrapper Helm charts for the Rook custom resources, besides the operators,
+being able to release the Helm charts independently of Rook releases makes it easy and quick to bring fixes to the Helm charts.
+This will decouple fixes to the current and upcoming Helm charts from the Rook releases and get them to the users as soon as possible.
+
+### Risks and Mitigation
```

Comment:
Currently we only have. "release" and "master" channels for the helm chart. Would that change or remain the same?

Reply:
That would change to only one channel as of right now.

Reply:
How would we test upcoming changes in master? Or how to test changes with each PR? For example, PRs frequently make updates to the helm chart to correspond to other changes in the PR. If the helm sources are in a different repo, we would have to separate the changes.

Reply:
A more advanced CI system such as [zuul](https://zuul-ci.org/) could help with this. But that obviously adds significant complexity so again I tend to think keeping them coupled is better.